src/Typedefs.idr

```diff
@@ -34,13 +35,21 @@ mutual
     ||| Application of a named type to a vector of arguments.
     TApp  : TNamed k -> Vect k (TDef n)           -> TDef n
 
+    ||| A Reference to a named typedef
+    TRef  : Name                                  -> TDef n
```

Comment:
For future consideration, I wonder if it would make sense to also have a scoped representation (ie an index of the datatype representing references) of these references in order to only represent and hence having to deal with legal states.